Implement automatic scene conversion

mphe opened the pull request:
This PR brings the foundation for running automatic version conversion on project startup.
Most importantly it implements scene conversion for the node type change to `MeshInstance2D` discussed in #172.
Also shortly before finishing this, Godot 4.4 was released, so I included the update to 4.4 as well, i.e. `*.uid` files and respective changes to the scenes.

The plugin checks if version information about SS2D exists in the project settings. If none was found or if it is different, all registered converter try to detect if changes are required.
If yes, a message will propup asking the user to confirm. The "OK" button is intentionally disabled for 3s to prevent users from confirming without reading.
When confirmed, all converters run sequentally and the success or failure is reported.
After success or when no conversion is required, the current version number is written to a custom internal project settings key to prevent having to run this procedure at every start.

That means the project conversion check will only run at project startup after initial install and after every version change until it was successful.
At the moment, the version number check is really just a `!=` comparison. I think it should suffice in most cases and if a more sophisticated check is required, it can be added later.

In the 3-dot "Extra Options" menu there is also another entry to re-run the project conversion check manually. This might be useful for people copying scenes from older projects to newer projects.

To make testing easier, I have not yet included a commit with the `MeshInstance2D` conversion applied.
I will do so after the PR was approved.

limbonaut commented on addons/rmsmartshape/plugin.gd:
```diff
@@ -1639,3 +1646,106 @@ func _debug_mouse_positions(mm: InputEventMouseMotion, t: Transform2D) -> void:
+	var success := true
+
+	for i in converters.size():
+		if converters[i].convert():
```
Shouldn't we check if converter needs to convert the project once again? I assume not every converter needs to run on version upgrade.

mphe replied:
I specified in the IVersionConverter class that if no conversion is necessary, `convert()` should do nothing.
https://github.com/SirRamEsq/SmartShape2D/blob/8da922eb7d0ac3c227e01fb511f760c8e1a6a5fa/addons/rmsmartshape/version_transition.gd#L15-L16

But for clarity and possible future changes, I can add another `needs_conversion()` check.

limbonaut commented on tests/unit/test_tscn_analyzer.gd:
```diff
@@ -0,0 +1,80 @@
+	var analyzer := _get_test_analyzer()
+	assert_true(analyzer.contains_shapes())
+
+	analyzer.load("res://tests/unit/res://tests/unit/test_convert_tscn_node_types.gd")
```
Path broken?

limbonaut commented on tests/unit/test_tscn_analyzer.gd:
```diff
@@ -0,0 +1,80 @@
+	assert_eq(analyzer._content_start_line, 11)
+
+	shape_ids.clear()
+	analyzer.load("res://tests/unit/res://tests/unit/test_convert_tscn_node_types.gd")
```
Same

limbonaut reviewed (commented):
Looks good so far. Are you going to create a separate PR for the MeshInstance2D implementation?

mphe commented:
Fixed.
Fixing the path actually revealed another little bug where the state of `TscnAnalyzer` was not properly reset on `load()`.
I also added the commit which includes the type change to `MeshInstance2D`.

> Are you going to create a separate PR for the MeshInstance2D implementation?

Yes, this PR is only for the conversion process.

limbonaut commented:
> Can also create a new PR for this if you want.

I'm fine with it being here. Do you want to keep this open until MeshInstance PR?
